GBPTree don't block close if need clean

burqen opened the pull request:
Close is not longer blocked by need cleaning. Instead, cleaner will just die silently if tree is closed. This is fine because the clean will just start again next time tree is opened.

Builds on #9309

Commit:
Close is not blocked by 'need clean'

It is fine to close the GBPTree even if it still needs cleaning.
The cleaning job will fail because the pagedFile is will be
closed, but that is fine. It will just start over next time
the tree is started.

Commit:
Test dying cleaner because of close

tinwelint commented on community/common/src/main/java/org/neo4j/kernel/impl/util/JobSchedulerAdapter.java:
```diff
@@ -0,0 +1,93 @@
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+public class JobSchedulerAdapter implements JobScheduler
```
This could live in src/test, couldn't it? In the vein of cutting down on implementations active at runtime.

tinwelint commented on community/index/src/main/java/org/neo4j/index/internal/gbptree/GroupingRecoveryCleanupWorkCollector.java:
```diff
@@ -30,29 +34,44 @@
         while ( (job = jobs.poll()) != null )
         {
-            job.run();
+            jobScheduler.schedule( recoveryCleanup, job );
```
I may be wrong, but it looks like we're scheduling all cleanup jobs individually. This also means that cleanup jobs could potentially run in parallel. What happens if there are multiple cleanup jobs, each using all available processors? There will be lots and lots of context switching in vain, won't there?

burqen replied:
Very good point and a nice catch! Addressing this in PR #9309

tinwelint commented:
Also there's a javadoc compilation issue in `GroupingRecoveryCleanupWorkCollector`

burqen commented:
@tinwelint Your comments a better suited for #9309 on which this PR builds.
Will address them there instead and rebase this branch.

MishaDemianenko commented:
@burqen base was merged so you can rebase now

Commit:
Fix test timeout

Because RecoveryCleanupWorkCollector was not
restarted together with LabelScanStore the
cleanup jobs was never triggered and the log
never saw the scan store recovery complete message.

Commit:
Remove unused variable
